### DISCUSSION TITLE
Pageview API data with PHP

### DISCUSSION THREAD

**Discussion by mayumih387** [umami-software/umami#881] [Q&A] [answered]
I'm trying to retrive pageviews with PHP.

I wrote following code but it doesn't work well. Could someone kindly help me?

```
class getRequest {
	public function curlJson( $url, $params = array() ) {
		if ( count( $params ) > 0 ) {
			$query = '?' . http_build_query( $params );
		} else {
			$query = '';
		}
		$url .= $query;
		
		$headers = array(
			'Content-type: application/json',
			'Cookie: umami.auth=eyJlbmMiO...Hnf3FoNF_4w',
		);

		if ( !function_exists( 'curl_version' ) ) {
			return false;
		}
			
		$ch = curl_init();
		
		curl_setopt( $ch, CURLOPT_URL, $url );
		curl_setopt( $ch, CURLOPT_TIMEOUT, 60 );
		curl_setopt( $ch, CURLOPT_RETURNTRANSFER, true );
		curl_setopt( $ch, CURLOPT_HTTPHEADER, $headers );
		$data = curl_exec( $ch );
		
		$response = curl_getinfo( $ch, CURLINFO_HTTP_CODE );
		
		curl_close( $ch );
		
		if ( $response === 200 ) {
			$data = json_decode( $data, true );
		} else {
			$data = false;
		};
		return $data;
	}
}
$request = new getRequest();
$url = 'https://my-umami-app.com/api/website/1/pageviews';
$params = array(
	'start_at' => strtotime('2021-11-01'),
	'end_at' => strtotime('2021-11-30'),
	'unit'=> 'day',
	'tz' => 'Asia/Tokyo'
);
$request->curlJson( $url, $params );
```

It returns `{"pageviews":[],"sessions":[]}` = empty.

Also empty with connecting from browser, i.e.:
`https://my-umami-app.com/api/website/1/pageviews?start_at=1635724800&end_at=1638230400&unit=day&tz=Asia%2FTokyo`

*But it worked with `/api/websites` (the data not requiring paramaters).

Thanks in advance!

**Comment by cywio** [comment, marked as answer]
You need to append 000 to the end of your unix timestamp since javascript uses milliseconds but php uses seconds, the API expects milliseconds

**Reply by mayumih387** [reply]
What a clear answer, it worked well!
Thank you so much!

**Reply by rimzzlabs** [reply]
> You need to append 000 to the end of your unix timestamp since javascript uses milliseconds but php uses seconds, the API expects milliseconds

any idea what is the timestamp mean? i passed the new Date().toISOString() to the start_date and end_date query parameters, but end up getting 500, but when I pass random 10 digits number, nothing goes wrong except I got an empty array of pageviews
